executor/include/execution/trait/can_prefer.hpp

```diff
@@ -0,0 +1,34 @@
+//
+// Created by Shrijit Singh on 17/06/20.
```

Comment:
Does it make sense to split `can_require` and `can_prefer` implementations? Especially since the implementation for both is exactly the same.

Reply:
That was a mistake they have slightly different implementations. It has been fixed in #12